### PR TITLE
backport: merge bitcoin#24241, #24534, #24948, #28622, #28880, #29185, #29170, #29233, #29298, #29598, #29732, #29890, #29739, #30074, #30198, #29072 (toolchain backports: part 2)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* Dependent on https://github.com/dashpay/dash/pull/6384
* Dependency for https://github.com/dashpay/dash/pull/6389
* The Qt patch introduced in [dash#5596](https://github.com/dashpay/dash/pull/5596), `fix_qt_placeholders.patch`, was a portion of a suggested workaround for QTBUG-92199 ([source](https://bugreports.qt.io/browse/QTBUG-92199?focusedId=669719&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-669719)) but since then, a fix ([here](https://codereview.qt-project.org/c/qt/qtbase/+/434310)) has made its way to 5.15.12 and we are upgrading to 5.15.14 from 5.15.11.

  So we can safely remove this patch.

## Breaking Changes

None expected

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation **(note: N/A)**
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Comment by kwvg** [comment]
### Expected hashes for 803520d

```
74ec7398193b50afec50c65fdb2c031d1d7c354f45240d8f3a987273e96a96b3  dashcore-22.0.0-beta.1-82-g803520d891d7-aarch64-linux-gnu-debug.tar.gz
cd18e8df22820551439cd52c315d4712cc738be5a65e7d5fa5e812b4a35443f7  dashcore-22.0.0-beta.1-82-g803520d891d7-aarch64-linux-gnu.tar.gz
2ba9112332d1d23393878c2152aff4d24e3bc63cbb36e3aa07705cfb0e4601fe  dashcore-22.0.0-beta.1-82-g803520d891d7-arm-linux-gnueabihf-debug.tar.gz
8147c8596da5eb18ea7190a8cf5e9c73770cff03694aa4d43389051ada73969f  dashcore-22.0.0-beta.1-82-g803520d891d7-arm-linux-gnueabihf.tar.gz
d3e83ea008d81dbc99c2f10f32865e2547567bea8faec5cca78add781c7208d2  dashcore-22.0.0-beta.1-82-g803520d891d7-arm64-apple-darwin-unsigned.tar.gz
5f550f04135117b7bea4a5d00d447f88b5712cc89737a7f22ea6906f84c1f1db  dashcore-22.0.0-beta.1-82-g803520d891d7-arm64-apple-darwin-unsigned.zip
cc6568b1b941a31fc9ead5d45f50d3948e451c719daa84d72c2e098b459bac75  dashcore-22.0.0-beta.1-82-g803520d891d7-arm64-apple-darwin.tar.gz
3d9fe43a8c60e94d262f36f1c8cd21e97bac94758dc1c4a4777034b76d8c4c88  dashcore-22.0.0-beta.1-82-g803520d891d7.tar.gz
12725898a43a063b979980f3ad0d074c2f77b6826e3e51cb88562b35f6c9a2ac  dashcore-22.0.0-beta.1-82-g803520d891d7-powerpc64-linux-gnu-debug.tar.gz
189bc107a3e516bf0d94a6c9fa16f0d575e5b28ae96b7a06ee25c22e44f21f26  dashcore-22.0.0-beta.1-82-g803520d891d7-powerpc64-linux-gnu.tar.gz
36debf5742c29b81eb0fd3a8c5b17e4739f3009e96947da7731343018d2e2722  dashcore-22.0.0-beta.1-82-g803520d891d7-riscv64-linux-gnu-debug.tar.gz
ecb43942495c3838c70f9cb049dfe0eaccd93de7c87e2eb42fa26cdbf75d2979  dashcore-22.0.0-beta.1-82-g803520d891d7-riscv64-linux-gnu.tar.gz
09e3200636ba2de39ac0014d213270dee684ac31ec240ba1d6ddd144600f0901  dashcore-22.0.0-beta.1-82-g803520d891d7-x86_64-apple-darwin-unsigned.tar.gz
e8fcc2326a9f6feb88e7b71431a1247fbbf740122440040ba9f82b9ede1db1c7  dashcore-22.0.0-beta.1-82-g803520d891d7-x86_64-apple-darwin-unsigned.zip
5c919427009ab3515f31792e2f15bb25cc966e8b976e196350d8514db0029ba1  dashcore-22.0.0-beta.1-82-g803520d891d7-x86_64-apple-darwin.tar.gz
82bc85e1e36c6977f1ae5046e2781937046fe39c6a6183bec5593acf82c20c1d  dashcore-22.0.0-beta.1-82-g803520d891d7-x86_64-linux-gnu-debug.tar.gz
99e727feb761ddef610ae8783a1a9cb82e719ae2f948ea079af01a3fcb695268  dashcore-22.0.0-beta.1-82-g803520d891d7-x86_64-linux-gnu.tar.gz
259fcc3ca0b7a6d3baa25c240c1300411c8138f69914a9fa40bcd883f5dba62d  dashcore-22.0.0-beta.1-82-g803520d891d7-win64-debug.zip
81a9f901e7d1d4200e03d1bcb69ff70b1a140e31f991ffefd2af425f20ab483e  dashcore-22.0.0-beta.1-82-g803520d891d7-win64-setup-unsigned.exe
8dc2ad3e17f30f84723369fdfbf9c6f74eb7abfacc0c61874ae6095c12bc9a92  dashcore-22.0.0-beta.1-82-g803520d891d7-win64-unsigned.tar.gz
c3551b046816d82e0688af538c116cdea72e187d80f3e2832a365e0d6e8c7705  dashcore-22.0.0-beta.1-82-g803520d891d7-win64.zip
```

**Comment by kwvg** [comment]
### Expected hashes for 18eeedd446e3

```
9c81077781dc22246fec6b0150f09e581fbab05eeb418b6d87c7e2303b1c36ab  dashcore-22.0.0-beta.1-82-g18eeedd446e3-aarch64-linux-gnu-debug.tar.gz
6a251dd11836c532fc3b123cd52ed36125f5749a647aba688b053bf5fe4e24bd  dashcore-22.0.0-beta.1-82-g18eeedd446e3-aarch64-linux-gnu.tar.gz
2f8b1ee8cd91d75eccf4b539f9b1e71cf5a71d68dab8f5dda134536aac9bd2e4  dashcore-22.0.0-beta.1-82-g18eeedd446e3-arm-linux-gnueabihf-debug.tar.gz
d02c94155e2aa06992a2b7a9acce1334c006ecdfa5db6761eb403858586a3a98  dashcore-22.0.0-beta.1-82-g18eeedd446e3-arm-linux-gnueabihf.tar.gz
8ed01ebd6c6fa3e6e1efb7a629c06fb830954a9c3d18ce143450cc3251d09d41  dashcore-22.0.0-beta.1-82-g18eeedd446e3-arm64-apple-darwin-unsigned.tar.gz
cafcd60aa54afbb0d80220acc21c3fef66d2257e4cbc30b693b8bfe3cee06b4b  dashcore-22.0.0-beta.1-82-g18eeedd446e3-arm64-apple-darwin-unsigned.zip
b95ecbfc44827e2a0016dab6e0ce8bee2d1d1a5945c6543e116b9c899adf3a1d  dashcore-22.0.0-beta.1-82-g18eeedd446e3-arm64-apple-darwin.tar.gz
ebaa689adfa33f78d5ec02aa6f0253f46dc2b02634f265135d356d699b3257da  dashcore-22.0.0-beta.1-82-g18eeedd446e3.tar.gz
b3487b4d43ddf53954a9d81abee387f2353db4f9701cf18d0818d0a7d4ac6d72  dashcore-22.0.0-beta.1-82-g18eeedd446e3-powerpc64-linux-gnu-debug.tar.gz
bb864611cdda82aa551cbf0f792a350e8e9d153b300496f525037a08e14349b6  dashcore-22.0.0-beta.1-82-g18eeedd446e3-powerpc64-linux-gnu.tar.gz
9357398bfc00697b61d5e476b336df8789d5f637834db343f3750be3b443643e  dashcore-22.0.0-beta.1-82-g18eeedd446e3-riscv64-linux-gnu-debug.tar.gz
3ddba86341685c47c83e40b65f57e66082075fe6807aba6c94b225bba6e71da5  dashcore-22.0.0-beta.1-82-g18eeedd446e3-riscv64-linux-gnu.tar.gz
a2b8ff232d2e5d1d1c6d294c34eba3e4a1d029858810a36fd855ab97fd7a9064  dashcore-22.0.0-beta.1-82-g18eeedd446e3-x86_64-apple-darwin-unsigned.tar.gz
c4880e9dea8d5ecd5c0c0b024eb294dbe741eeb0def69430843f9036d28a0f82  dashcore-22.0.0-beta.1-82-g18eeedd446e3-x86_64-apple-darwin-unsigned.zip
22912168876bf159c8f818c4d86268aad44d97edf0394e7adecb9ce1039ca14f  dashcore-22.0.0-beta.1-82-g18eeedd446e3-x86_64-apple-darwin.tar.gz
b9bf85f6688ddacdb109ac2196f8d0d5d151464249d11c1f6c1472cde29b1a5a  dashcore-22.0.0-beta.1-82-g18eeedd446e3-x86_64-linux-gnu-debug.tar.gz
e07f194c1aef7434aed7d50dfdf60987567ac069b0e4753aa145b981750bd4c6  dashcore-22.0.0-beta.1-82-g18eeedd446e3-x86_64-linux-gnu.tar.gz
05ccc04c6bac83bde43765bf58d117b48f5b3c663a6aeb6d0354b064bb107706  dashcore-22.0.0-beta.1-82-g18eeedd446e3-win64-debug.zip
6bb51e8b3dbf2add1de9cc94927a292328c7881da770b6422e1b272382e1a410  dashcore-22.0.0-beta.1-82-g18eeedd446e3-win64-setup-unsigned.exe
32d36387b02a6903dc327aa2a648fb681495c1ba56f504725fc1ee8ec0db08fb  dashcore-22.0.0-beta.1-82-g18eeedd446e3-win64-unsigned.tar.gz
73896ea269228ab8e2650cb857a9ad51f5d09fbf5ff92c48321f51d01d2644ae  dashcore-22.0.0-beta.1-82-g18eeedd446e3-win64.zip
```

**Comment by kwvg** [comment]
### Expected hashes for f5ff9280

```
2d72e55854cd0db064ea4dab37b9ec8a11c2c57aca235e8c8fd0856cb52363df  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-aarch64-linux-gnu-debug.tar.gz
9ed796415aca45e438837d9abbbe7d8179209007cae491398bbaccfdcbece7c8  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-aarch64-linux-gnu.tar.gz
58b7d2788332a3794859733e8696eae76fa7180d3060438b1d0d876f91ca9288  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-arm-linux-gnueabihf-debug.tar.gz
311f87da7cdb1d9bf58262000d7cd71bbeeca87b3b0a97f543841082d02a6517  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-arm-linux-gnueabihf.tar.gz
52e9936e6da09ccfa7504dc9617e1c83678be16c9e02e5f96c85349f7fabfcde  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-arm64-apple-darwin-unsigned.dmg
8b3c1e0e4cf15b6bd76c9518cdcf3eb70deeb381c8983afc4ec2349422c7f345  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-arm64-apple-darwin-unsigned.tar.gz
8f20664565f1c32062aa94a4ef4f5adff8f18ff702545de5cd3b000c4d755ea6  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-arm64-apple-darwin.tar.gz
95d0be1b1d46c978dd4172f7a71da6d779a491c359a1ee6c9e7d1f6622230221  dashcore-22.0.0-rc.1-14-gf5ff92807dc8.tar.gz
67223da62ab7cabfba7c31f859af67338c0ee1b3ec31af1a402f21bca19f4c03  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-powerpc64-linux-gnu-debug.tar.gz
02fde6bf17de629c904cb3060db8fe54fe663bcce6c2aa3e09d0f730e268cd19  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-powerpc64-linux-gnu.tar.gz
6c10a26b49ec1c80ff698455bc17a157e26e709e4c6fbd45704b151c7710a5cd  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-riscv64-linux-gnu-debug.tar.gz
9957aa20c676be0ee791252e21e966f0b227192f280dd77ad6f643050d7bbd84  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-riscv64-linux-gnu.tar.gz
37e0b730639420b89e4df8cc157697a2a6b419926ef62427c84de2c69243e8e2  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-x86_64-apple-darwin-unsigned.dmg
4d3e76cc9beb8458b3adead79fbf5a0b7488d5e0370dde576d8df074d1a3c1f4  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-x86_64-apple-darwin-unsigned.tar.gz
229be1ef698f582c3a1a97b5c11af5c68e353ed5dbb2e0e17faccc5e20d23d8a  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-x86_64-apple-darwin.tar.gz
d6e7ba93f9e096be9a29f136146f0b012f3eddad8fa733dea0629fe44c6dea19  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-x86_64-linux-gnu-debug.tar.gz
c48ba4a0b454901f70a734bda6a8ecefff52a5d7694cb8775cd069397b78b159  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-x86_64-linux-gnu.tar.gz
13acb18c4657e95da3bdc88693cda031cb5374b7c189c770ed24acea69ddb70b  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-win64-debug.zip
44cf9591b089ac94b6220065141a9df9fa45c01f2708b88442d0b7623c019b28  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-win64-setup-unsigned.exe
f2ebaf6fcdef34c6f3c8220ec609f1949197898dace045808e9bbacdebcd69fd  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-win64-unsigned.tar.gz
6cba5aeee7b6fae45ecaf6d92d9b6d51471e1ee8d25615e7371136f0b88b0983  dashcore-22.0.0-rc.1-14-gf5ff92807dc8-win64.zip
```

File: contrib/macdeploy/README.md [hunk 1]
```diff
@@ -12,47 +12,48 @@ When complete, it will have produced `Dash-Qt.dmg`.
 
 ### Step 1: Obtaining `Xcode.app`
 
-Our current macOS SDK
-(`Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers.tar.gz`) can be
-extracted from
-[Xcode_12.2.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.2/Xcode_12.2.xip).
+A free Apple Developer Account is required to proceed.
```
[the remark this answers is not in the document]
Comment (kwvg) [reply contrib/macdeploy/README.md:15]: Resolved in latest push!

**Review by UdjinM6 — APPROVED** [review]
utACK d777c7d7ff6d2decb21a8812cbddd90de4d2001f

File: depends/README.md [hunk 2]
```diff
@@ -126,7 +126,6 @@ The following can be set when running make: `make FOO=bar`
 - `LOG`: Use file-based logging for individual packages. During a package build its log file
   resides in the `depends` directory, and the log file is printed out automatically in case
   of build error. After successful build log files are moved along with package archives
-- `LTO`: Use LTO when building packages.
```
Comment (PastaPastaPasta) [inline depends/README.md:129]: 29185 upstream kept a line in? https://github.com/bitcoin/bitcoin/pull/29185/files#diff-dff7e06f00d3fb1cd0ccf1a491314d52457870824b4cbabfdc4a8ca77ccd7586R126

File: depends/packages/qt.mk [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
-$(package)_sha256_hash=425ad301acd91ca66c10c0dabee0704e2d0cd2801a6b670115800cbb95f84846
+$(package)_sha256_hash=4cca51dcc1f22ceeee6b3e33cd1c3a60b14e85e24644dca3af89a2c2989ab809
 $(package)_linux_dependencies=freetype fontconfig libxcb libxkbcommon libxcb_util libxcb_util_render libxcb_util_keysyms libxcb_util_image libxcb_util_wm
 $(package)_qt_libs=corelib network widgets gui plugins testlib
 $(package)_linguist_tools = lrelease lupdate lconvert
 $(package)_patches = qt.pro
 $(package)_patches += qttools_src.pro
 $(package)_patches += mac-qmake.conf
 $(package)_patches += fix_qt_pkgconfig.patch
-$(package)_patches += fix_qt_placeholders.patch
```
Comment (PastaPastaPasta) [inline depends/packages/qt.mk:14]: this is extra from upstream commit?

**Review by PastaPastaPasta — COMMENTED** [review]
generally LGTM d777c7d7ff6d2decb21a8812cbddd90de4d2001f, few concerns

File: depends/packages/qt.mk (hunk 3 above)
Comment (kwvg) [reply depends/packages/qt.mk:14]: As mentioned in the PR description, `fix_qt_placeholders.patch` was a patch introduced by [dash#5596](https://github.com/dashpay/dash/pull/5596) and no longer necessary as the problem that was addressing has been resolved in the version of Qt that we're bumping to.

**Comment by kwvg** [comment]
### Expected hashes for 1506d9d

```
548394d2e50cfb9a862fad5081b819a4200d2f0260c436507d32e98a636aa6f5  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-aarch64-linux-gnu-debug.tar.gz
213b86ae5fccc6fb992167207d0416db1a98611d06ec00ffd063210ff9b88d12  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-aarch64-linux-gnu.tar.gz
996a345b57479204d9e002ca3153ae220ca536912c44eb8afb29ab4b6e2edb42  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-arm-linux-gnueabihf-debug.tar.gz
ea4720e79097faa50de10def662243ef56debb4a69b2b70c2a3d0371f9c5eea9  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-arm-linux-gnueabihf.tar.gz
76370201fd44407b184c8730e02de856f174efaf34b9d45ad2be1dbe14aa055e  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-arm64-apple-darwin-unsigned.dmg
200aabc259001c1aaacb23b2dc8bc7ce9f6bf23b1f90af5f96157947a57b1f1b  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-arm64-apple-darwin-unsigned.tar.gz
f848a2667742781a5b4b4f12fc2d681864e8cc1e657d5068d8cf4cc4492037ad  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-arm64-apple-darwin.tar.gz
3be96c0b1fd9d102a6700289ff050dce96d972a43a05988fc75fc15fa46b4540  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4.tar.gz
540f1686942ead5e1b65c9db87c50d66546ec93154d4bd03762ac835a4948e64  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-powerpc64-linux-gnu-debug.tar.gz
298a4eb3394228eb28ec9cdba0c4110b3afd4dfa8a870b53f7726f1e6cddc9f8  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-powerpc64-linux-gnu.tar.gz
ca3eeb0f1b0b08eced7ef91700802015a259df937f4628ad43a81e4e535c1771  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-riscv64-linux-gnu-debug.tar.gz
7707a425fdbf4951be3387f95dcf90b8262dfb58260b04d4ea8c7016f6af9a5a  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-riscv64-linux-gnu.tar.gz
2231ec5a2fc10de47ee435dd995feb257968136a094a1c1a67c1c9dc2030bca0  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-x86_64-apple-darwin-unsigned.dmg
494672f308865bf0692dabc19227615e1ed8cc78d277640b9ec4e8f97982abda  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-x86_64-apple-darwin-unsigned.tar.gz
3ab3d78db314148fecbb8f062fee7ba1b9787bb03411f84d6514512425d2e8e0  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-x86_64-apple-darwin.tar.gz
c50eef9a0abe2de18bf775bb28457431c78bbd9ef4dab59a47df9193ff00baa4  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-x86_64-linux-gnu-debug.tar.gz
e356151aac9536d8f7eebd2c1c28d6363e3ae55896babe091806438a4192712e  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-x86_64-linux-gnu.tar.gz
c9fab2b10a6b800891234e5c2253351818f581de7b82bf8a16f69f30929a3e03  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-win64-debug.zip
527e958edab39cf4a0f8a34c41574c7df0bc55e2b163e565ca9365644f8f99f5  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-win64-setup-unsigned.exe
9b02dd62956737db71afcb023feefdb86585afa0539b94c8ba46679a0ef4d373  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-win64-unsigned.tar.gz
97f42b6913f4676417c0efbb199f3a5d1ac5d0a2ac8ccf580f0c6e0737d8385f  dashcore-22.0.0-rc.1-18-g1506d9d9b8d4-win64.zip
```

File: depends/README.md (hunk 2 above)
Comment (kwvg) [reply depends/README.md:129]: Resolved in latest push!

**Review by UdjinM6 — APPROVED** [review]
utACK 1506d9d9b8d456f866928b06666ee7bd49a26562

**Review by PastaPastaPasta — APPROVED** [review]
utACK 1506d9d9b8d456f866928b06666ee7bd49a26562

**Pull request merged**